### ISSUE TITLE
tensorflow:Gradients do not exist for variables 

### ISSUE THREAD

**Issue by LoveFishoO** [tensorflow/gnn#93]
When I try to change the feature name from hidden_state to other, i meet a mistake. The program shows that WARNING:tensorflow:Gradients do not exist for variables ['gnn/essay_model/graph_update/edge_set_update/next_state_from_concat/dense_3/kernel:0', 'gnn/essay_model/graph_update/edge_set_update/next_state_from_concat/dense_3/bias:0', 'gnn/essay_model/graph_update/node_set_update/next_state_from_concat_1/dense_4/kernel:0', 'gnn/essay_model/graph_update/node_set_update/next_state_from_concat_1/dense_4/bias:0'] when minimizing the loss. If you're using `model.compile()`, did you forget to provide a `loss`argument?

following is error code:
```
node_set = tfgnn.NodeSet.from_fields(
                sizes=[node_num],
                features={
                    'node_feature': node_feature, 
                }
            )

edge_set = tfgnn.EdgeSet.from_fields(
                sizes=[edge_num],
                features={
                    'edge_feature': tf.convert_to_tensor(edge_feature, dtype=tf.float32)},
                adjacency=tfgnn.Adjacency.from_indices(
                    source=('node', adjacency[:, 0]),
                    target=('node', adjacency[:, 1])
                ),
            )

context = tfgnn.Context.from_fields(features=None)

graph_tensor = tfgnn.GraphTensor.from_pieces(
                node_sets={'node': node_set},
                edge_sets={'edge': edge_set},
                context=context,
            )



model = tfgnn.keras.layers.GraphUpdate(
            edge_sets={
                "edge": tfgnn.keras.layers.EdgeSetUpdate(
                    edge_input_feature=['edge_feature'],
                    node_input_feature='node_feature',
                    next_state=tfgnn.keras.layers.NextStateFromConcat(tf.keras.layers.Dense(2)))
            },
            node_sets={
                "node": tfgnn.keras.layers.NodeSetUpdate(
                    edge_set_inputs={"edge": tfgnn.keras.layers.Pool(tfgnn.TARGET, "sum")},
                    node_input_feature='node_feature',
                    next_state=tfgnn.keras.layers.NextStateFromConcat(
                        tf.keras.layers.Dense(264)
                    ),
                )
            },
        )

```

however, when i use the hidden_state as the feature name, i will not meet this question. following is code

```
node_set = tfgnn.NodeSet.from_fields(
                sizes=[node_num],
                features={
                    'hidden_state': node_feature, 
                }
            )

edge_set = tfgnn.EdgeSet.from_fields(
                sizes=[edge_num],
                features={
                    'hidden_state': tf.convert_to_tensor(edge_feature, dtype=tf.float32)},
                adjacency=tfgnn.Adjacency.from_indices(
                    source=('node', adjacency[:, 0]),
                    target=('node', adjacency[:, 1])
                ),
            )

context = tfgnn.Context.from_fields(features=None)

graph_tensor = tfgnn.GraphTensor.from_pieces(
                node_sets={'node': node_set},
                edge_sets={'edge': edge_set},
                context=context,
            )



model = tfgnn.keras.layers.GraphUpdate(
           
     edge_sets={
                "edge": tfgnn.keras.layers.EdgeSetUpdate(
                    next_state=tfgnn.keras.layers.NextStateFromConcat(tf.keras.layers.Dense(2)))
            },
            node_sets={
                "node": tfgnn.keras.layers.NodeSetUpdate(
                    edge_set_inputs={"edge": tfgnn.keras.layers.Pool(tfgnn.TARGET, "sum")}
                    next_state=tfgnn.keras.layers.NextStateFromConcat(
                        tf.keras.layers.Dense(264)
                    ),
                )
            },
        )


```

I do not understand why this question happend. And how could i solve this question? 

I am forward to your help. Thank you!
